### PR TITLE
Implement default join alias via `@MappedProperty`

### LINKED ISSUE #1705: Support `@MappedProperty(alias="abc")` as a default join alias
### Feature description

There is an existing feature where we will support a property alias in the resulting SELECT cause, it would be nice to support the alias on the joined associations as well.
```java
class Parent {
    @Relation(...)
    Child child
}
```
Will produce a default join alias `parent_child_` when `@Join` is used without an explicit alias.

The idea is to override the default alias:
```java
class Parent {
    @MappedProperty(alias="abc")
    @Relation(...)
    Child child
}
```

In that case, the method`@Join("child") List findAll()` should generate the same SQL query as the method `@Join(value="child", alias="abc") List findAll`:
`SELECT abc.'some_column_from_child',... FROM PARENT JOIN CHILD ...`

### PR DESCRIPTION
**Pull request by Goldmensch**
closes #1705 

The pr probably contains breaking changes in the query generation if a `@MappedProperty` was already used.

### CODE REVIEW

File: data-processor/src/main/java/io/micronaut/data/processor/visitors/RepositoryTypeElementVisitor.java [hunk 1]
```diff
@@ -252,6 +254,28 @@ public void visitMethod(MethodElement element, VisitorContext context) {
                         entityResolver
                 );
 
+                // choose right join alias
+                // ignore pageable methods because they don't support join aliases
+                if (element.hasAnnotation(JoinSpecifications.class) && !methodMatchContext.hasParameterInRole(TypeRole.PAGEABLE)) {
```
Comment (dstepanov) [inline data-processor/src/main/java/io/micronaut/data/processor/visitors/RepositoryTypeElementVisitor.java:259]: The logic should be implemented in `AbstractSqlLikeQueryBuilder`: `getAliasName` or `getPathOnlyAliasName`.

**Review by dstepanov — CHANGES_REQUESTED** [review]

**Comment by dstepanov** [comment]
Please fix the tests

**Comment by Goldmensch** [comment]
Tests are passing locally, only checkStyle is violated due to the file length fo `SqlQueryBuilder` and `AbstractSqlLikeQueryBuilder`

**Comment by dstepanov** [comment]
Try to annotate the class with `@SuppressWarnings("FileLength")` try if it passes by running `./gradlew checkstyleMain`

**Comment by Goldmensch** [comment]
The annotations didn't work, so I added a filename based suppression for the file length check.

**Comment by dstepanov** [comment]
Thanks!

**Pull request merged**